test/jdk/javax/swing/Popup/TaskbarPositionTest.java

```diff
@@ -209,25 +214,38 @@ private void maybeShowPopup(MouseEvent e) {
         }
     }
 
+    // for debugging purpose, saves screen capture when test fails.
+    private static void saveScreenCapture(Rectangle bounds) {
+        BufferedImage image = robot.createScreenCapture(bounds);
+        try {
+            ImageIO.write(image,"png", new File("Screenshot.png"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
     /**
      * Tests if the popup is on the screen.
      */
     private static void isPopupOnScreen(JPopupMenu popup, Rectangle checkBounds) {
         if (!popup.isVisible()) {
+            saveScreenCapture(checkBounds);
             throw new RuntimeException("Popup not visible");
         }
```

Comment:
I don't really like this approach, you're modifying several places where an exception can be thrown, but there are other places.

I suggest adding a catch section into the `main` method:

```java
            robot.waitForIdle();
        } catch (Throwable t) {
            saveScreenCapture(screens);
            throw t;
        } finally {
            SwingUtilities.invokeAndWait(() -> {
```

This way you capture screenshot in one place and it automatically covers *all the cases* where an error is thrown, which avoids adding explicit calls to save the screenshots.

You may pass `robot` too.

I think you should make a screenshot of all the screens.